plugins/discourse-local-dates/test/javascripts/integration/rich-editor-extension-test.js

```diff
@@ -0,0 +1,38 @@
+import { module, test } from "qunit";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+import { testMarkdown } from "discourse/tests/helpers/rich-editor-helper";
+
+module(
+  "Integration | Component | prosemirror-editor - local-dates plugin extension",
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    Object.entries({
+      "local date": [
+        "[date=2021-01-01 time=12:00:00]",
+        '<p><span class="discourse-local-date cooked-date" data-date="2021-01-01" data-time="12:00:00" contenteditable="false">2021-01-01 12:00:00</span></p>',
+        "[date=2021-01-01 time=12:00:00]",
+      ],
+      "local date with timezone": [
+        '[date=2021-01-01 time=12:00:00 timezone="America/New_York"]',
+        '<p><span class="discourse-local-date cooked-date" data-date="2021-01-01" data-time="12:00:00" data-timezone="America/New_York" contenteditable="false">2021-01-01 12:00:00</span></p>',
+        '[date=2021-01-01 time=12:00:00 timezone="America/New_York"]',
+      ],
+      "local date range": [
+        "[date-range from=2021-01-01 to=2021-01-02]",
+        '<p><span class="discourse-local-date-wrapper" contenteditable="false"><span class="discourse-local-date cooked-date" data-range="from" data-date="2021-01-01">2021-01-01</span> → <span class="discourse-local-date cooked-date" data-range="to" data-date="2021-01-02">2021-01-02</span></span></p>',
+        "[date-range from=2021-01-01 to=2021-01-02]",
+      ],
+      "local date range with time": [
+        '[date-range from=2021-01-01T12:00:00 to=2021-01-02T13:00:00 timezone="America/New_York"]',
```

Comment:
This bbcode is not parsing for me on paste, or at least not adding the underline/click for the timezone list. Top is after pasting it in the old markdown composer, bottom is after pasting into rich editor:

![image](https://github.com/user-attachments/assets/890292dd-9158-4d14-afd2-fb5eb0a482fa)

Reply:
You mean when you paste html from cooked or from the editor itself? If so, yeah it's tricky because it contains two different elements without a container... or are you experiencing this in another scenario?

Reply:
No if i paste this BBCode `[date-range from=2021-01-01T12:00:00 to=2021-01-02T13:00:00 timezone="America/New_York"]` into the editor

Reply:
Thanks, it should be fixed now!